### ISSUE TITLE
Follow Arduino Library Guidelines

### ISSUE THREAD

**Issue by cisasteelersfan** [M-Hawkins/M3-LS_Micromanipulator#4]
We may want to consider restructuring the library to conform with Arduino [library guidelines](https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification). I was installing this on Eric's Windows laptop, and I had to:

Rename M3LS.cc to M3LS.cpp
Symlinks don't work in Windows so it would be best to move M3LS.cpp/h, hidjoystickrptparser.cpp/h to /src and symlink them to where we need to edit them.
Move examples to root directory.

Basically we need to completely reorganize this if we want to be able to install via the Sketch-> Include Library -> Add .ZIP Library (which I think we should strive for).

**Comment by M-Hawkins** [comment]
I think a lot of our headache stems from attempting to make our development and release directories one and the same. Creating the separate symlink'd folder worked for then, but as you pointed out that isn't a universal solution. 
In its current state, this repo is not set up for the Arduino library alone, but for all code related to the project. I believe this is the right way to go, especially for any summer work, as that may involve GUI development. 
I think we should look into pursuing two routes. The first would be how to prepare releases that provide only the necessary files. This would be a download for the library that doesn't contain any tests, mocks, or python code, and follows the library guidelines that you've linked too. This enables potential users to download and install the library quickly and painlessly, as you have outlined. It also allows us to release components separately, which may become more relevant down the line. 
This make this release process easier, we should outline instructions to build a release from source. Automation of this process would also be ideal. While definitely not the simplest solution, I believe this would help to continue facilitating test driven development going forward.

**Comment by cisasteelersfan** [comment]
I agree! We could even create our own release using the [release](https://github.com/M-Hawkins/M3-LS_Micromanipulator/releases/new) feature of Github (which I've never created myself, but I've downloaded from before).
